docs/sc-memory/api/cpp/extended/permissions_api.md

```diff
@@ -0,0 +1,809 @@
+    </tr>
+  </tbody>
+</table>
+
+### **Global and local user permissions**
+
+Depending on where the user can perform actions in the knowledge base, their permissions are divided into global and local permissions:
+
+* global permissions is permissions over entire knowledge base;
+* local permissions is permissions in some part (sc-structure) of knowledge base.
+
+Permissions can be set on a specific user or group (set or class) of users.
+
+All permissions can be represented as connections between users, action classes and sc-structures:
+
+<table>
+  <thead>
+    <tr>
+      <th>Permissions class</th>
+      <th>System identifier of relation sc-node for concrete user</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>Global permissions</td>
+      <td><code>nrel_user_action_class</code></td>
+      <td>Binary relation between concrete user and action class</td>
+    </tr>
+    <tr>
+      <td>Local permissions</td>
+      <td><code>nrel_user_action_class_within_sc_structure</code></td>
+      <td>Ternary relation between concrete user and action class with sc-structure where actions of specified class can be performed by this user</td>
+    </tr>
+  </tbody>
+</table>
+
+<table>
+  <thead>
+    <tr>
+      <th>Permissions class</th>
+      <th>System identifier of relation sc-node for users set</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>Global permissions</td>
+      <td><code>nrel_users_set_action_class</code></td>
+      <td>Binary relation between users set and action class</td>
+    </tr>
+    <tr>
+      <td>Local permissions</td>
+      <td><code>nrel_users_set_action_class_within_sc_structure</code></td>
+      <td>Ternary relation between users set and action class with sc-structure where actions of specified class can be performed by this one of users</td>
+    </tr>
+  </tbody>
+</table>
+
+Examples of global and local read permissions for user and users set:
+
+<table>
+  <tr>
+    <th>Permissions</th>
+    <th>SCg-code example</th>
+    <th>SCs-code example</th>
+  </tr>
+
+  <tr>
+    <td>User has global read permissions</td>
+    <td><scg src="../images/permissions/global_read_permissions_for_user.gwf"></scg></td>
+    <td>
+      <pre>
+        <code class="js hljs javascript">
+..user <~ concept_user;;
+
+nrel_user_action_class
+~> (..user => action_read_from_sc_memory);; 
+        </code>
+      </pre>
+    </td>
+  </tr>
+  <tr>
+    <td>User has local read permissions</td>
+    <td><scg src="../images/permissions/local_read_permissions_for_user.gwf"></scg></td>
+    <td>
+      <pre>
+        <code class="js hljs javascript">
+..user <~ concept_user;;
+
+..structure = [*
+    concept_set ~> ..set;;
+*];;
+
+nrel_user_action_class_within_sc_structure
+~> (..user => (action_read_from_sc_memory => ..structure));;
+        </code>
+      </pre>
+    </td>
+  </tr>
+  <tr>
+    <td>Each user in users set has global read permissions</td>
```

Comment:
Need info how to create new users set. Is it any set that has outgonig sc-arc to user? Or set that has relation nrel_inclusion with concept_set